### PR TITLE
feat: Review and create new ITF unittests

### PR DESCRIPTION
**Pull request by TedRio**
Add missing unittests, coverage is now at 90%

### CODE REVIEW

**Review by sebclrsn — APPROVED** [review]
As long as these new tests don't generate new warnings (regarding Julien's PR to delete them all), looks good to me

File: docs/whats_new/version_0_16_0.rst [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+
+Fail Fast
+^^^^^^^^^
+
+With the new --fail-fast flag which can be passed through the pykiso cli,
+tests will be stopped on the first error or failure.
+
+At the same time the behavior of the auxiliary create instance was changed.
+When the auxiliary instance creation are now failing, they will raise an exception.
+Combined with the new flag the test execution will be stopp when something went wrong.
```
Comment (kakooloukia) [inline docs/whats_new/version_0_16_0.rst:22]:
```suggestion
Combined with the new flag the test execution will be stopped when something go wrong.
```

**Review by kakooloukia — CHANGES_REQUESTED** [review]

**Review by sebastianpfischer — APPROVED** [review]

**Comment by sebastianpfischer** [comment]
commit convention is also not fine

**Review by kakooloukia — APPROVED** [review]

**Review by BKaDamien — APPROVED** [review]

**Review by Pog3k — APPROVED** [review]

File: tests/test_socketcan_to_trc.py [hunk 2]
```diff
@@ -199,3 +199,41 @@ def test_on_message_received(caplog, mocker, tmp_file):
...
+)
+def test_get_type(
+    is_remote_frame_value,
+    is_error_frame_value,
+    is_fd_value,
+    bitrate_switch_value,
+    error_state_indicator_value,
+    expected_return,
+):
+    class access_static_method(SocketCan2Trc):
```
Comment (Pog3k) [inline tests/test_socketcan_to_trc.py:224]: Class is here is not necessary.

Static methods can be directly called.

SocketCan2Trc.get_type(...)

File: tests/test_socketcan_to_trc.py [hunk 3]
```diff
@@ -199,3 +199,41 @@ def test_on_message_received(caplog, mocker, tmp_file):
...
+
+    can_msg = can.Message(
+        is_remote_frame=is_remote_frame_value,
+        is_error_frame=is_error_frame_value,
+        is_fd=is_fd_value,
+        bitrate_switch=bitrate_switch_value,
+        error_state_indicator=error_state_indicator_value,
+    )
+
+    assert access_static_method().get_type(can_msg) == expected_return
```
Comment (Pog3k) [inline tests/test_socketcan_to_trc.py:239]:
```suggestion
    assert SocketCan2Trc.get_type(can_msg) == expected_return
```

File: tests/test_simulated_auxiliary.py [hunk 4]
```diff
@@ -95,6 +105,127 @@ def prepare_config():
...
+        _cc_close = mocker.stub(name="_cc_close")
+        _cc_send = mocker.stub(name="_cc_send")
+        _cc_receive = mocker.stub(name="_cc_receive")
+
+    return MockProxyCChannel(name="test_aux")
+
+
+@pytest.fixture
+def simulated_constructor_init(mocker, mock_channel, mock_msg):
+    class Dict_modified(dict):
```
Comment (Pog3k) [inline tests/test_simulated_auxiliary.py:127]: class not necessary ?
Pytest gives you the tools.
mocker.patch.object(simulation.Simulation, "get_scenario", return_value=scenario.Scenario([None]))

File: tests/test_simulated_auxiliary.py [hunk 5]
```diff
@@ -95,6 +105,127 @@ def prepare_config():
...
+
+        def get_scenario(self, test_suite_id: int, test_case_id: int):
+            return scenario.Scenario([None])
+
+    mocker.patch.object(simulation.Simulation, "__init__", return_value=None)
+    proxy_inst = simulated_auxiliary.SimulatedAuxiliary(
+        name="test_aux",
+        com=mock_channel,
+    )
+    proxy_inst.context = Dict_modified()
```
Comment (Pog3k) [inline tests/test_simulated_auxiliary.py:139]:
```suggestion
    mocker.patch.object(simulation.Simulation, "get_scenario", return_value=scenario.Scenario([None]))
```

File: tests/test_simulation.py [hunk 6]
```diff
@@ -0,0 +1,228 @@
...
+
+        assert isinstance(simulation_inst.map_context[1, 6], Scenario)
+        assert simulation_inst.map_context[0, 0].pop(0) == "handle_ping_pong"
+        assert simulation_inst.map_context[1, 1].pop(0) == "handle_successful"
+        assert simulation_inst.map_context[1, 2].pop(0) is True
+        assert simulation_inst.map_context[1, 3].pop(0) == "Test_implemented"
+        assert simulation_inst.map_context[1, 4].pop(0) is False
+        assert simulation_inst.map_context[1, 5].pop(0) == "Test_report"
+        assert simulation_inst.map_context[1, 6].pop(0) == "Test_report_run_failed"
+        assert simulation_inst.map_context[1, 7].pop(0) == "Test_reprun_not_implemented"
```
Comment (Pog3k) [inline tests/test_simulation.py:174]:
```suggestion
        assert simulation_inst.map_context[1, 7].pop(0) == "Test_report_run_not_implemented"
```

File: tests/test_simulation.py [hunk 7]
```diff
@@ -0,0 +1,228 @@
...
+        assert (
+            simulation_inst.map_context[1, 8].pop(0) == "Test_lost_com_during_run_ack"
+        )
+        assert (
+            simulation_inst.map_context[1, 9].pop(0) == "Test_lost_com_during_run_rep"
+        )
+        assert simulation_inst.map_context[1, 10].pop(0) == "Test_report_Teardown"
+        assert (
+            simulation_inst.map_context[1, 11].pop(0) == "Test_notipml_report_teardown"
```
Comment (Pog3k) [inline tests/test_simulation.py:183]:
```suggestion
            simulation_inst.map_context[1, 11].pop(0) == "Test_not_impl_report_teardown"
```

File: tests/test_proxy_auxiliary.py [hunk 8]
```diff
@@ -175,49 +201,44 @@ def test_check_compatibility_exception(mocker, cchannel_inst):
...
+        (None, False, True),
+        (ValueError(), True, False),
+    ],
+)
+def test_create_auxiliary_instance_valid(
+    mocker, cchannel_inst, side_effect_value, stop_event_set, expected_function_return
+):
     mocker.patch.object(ProxyAuxiliary, "run")
-    mocker.patch.object(CChannel, "open", side_effect=ValueError())
+    mocker.patch.object(CChannel, "open", side_effect=side_effect_value)
```
Comment (Pog3k) [inline tests/test_proxy_auxiliary.py:215]: name could give more context - for example
```suggestion
    mocker.patch.object(CChannel, "open", side_effect=open_return_value)
```

**Review by Pog3k — COMMENTED** [review]
see my comments

**Pull request merged**